packages/financial-templates-lib/src/price-feed/ExpressionPriceFeed.js

```diff
@@ -0,0 +1,154 @@
+const assert = require("assert");
+const { PriceFeedInterface } = require("./PriceFeedInterface");
+const Web3 = require("web3");
+const { create, all } = require("mathjs");
+
+// Customize math (will be exported for other modules to use).
+const math = create(all, { number: "BigNumber", precision: 100 });
+const nativeIsAlpha = math.parse.isAlpha;
+const allowedSpecialCharacters = Array.from("[]/ -");
+
+// Modifies math.js's function to determine whether a character is allowed in a symbol name.
+math.parse.isAlpha = function(c, cPrev, cNext) {
+  // This character is the escape and the next is the special character.
+  const isValidEscapeChar = c === "\\" && allowedSpecialCharacters.includes(cNext);
+
+  // This character is the special character and the previous is the escape.
+  const isSpecialChar = cPrev === "\\" && allowedSpecialCharacters.includes(c);
+
+  return nativeIsAlpha(c, cPrev, cNext) || isValidEscapeChar || isSpecialChar;
+};
+
+// This escapes all the special characters (defined in the allowedSpecialCharacters array) in a string.
+// For example (note that "\\" is the js representation for a single literal backslash)
+// "ab/c d [e]" -> "ab\\/c\\ d\\[e\\]"
+function escapeSpecialCharacters(input) {
+  return Array.from(input)
+    .map((char, index, array) => {
+      if (allowedSpecialCharacters.includes(char) && array[index - 1] !== "\\") {
+        return `\\${char}`;
+      } else {
+        return char;
+      }
+    })
+    .join("");
+}
```

Comment:
this function looks tricky, you might want a test for this, though im pretty sure its working by proxy of testing everything else

Reply:
It has a test (not sure if this was true when you reviewed).

Reply:
Sorry, I just now realized that I forgot to push a few changes, but there's definitely a test in there now :)